db_test.go

```diff
@@ -425,6 +426,52 @@ func (s *SurrealDBTestSuite) TestSmartMarshalQuery() {
 	})
 }
 
+func (s *SurrealDBTestSuite) TestConcurrent() {
+	var wg sync.WaitGroup
+	totalcoroutine := 100
+
+	user := testUser{
+		Username: "electwix",
+		Password: "1234",
+	}
+
+	s.Run(fmt.Sprintf("%d Request Concurrent select n-exists", totalcoroutine), func() {
+		for i := 0; i < totalcoroutine; i++ {
+			wg.Add(1)
+			go func(j int) {
+				_, err := s.db.Select(fmt.Sprintf("users:%d", j))
+				s.Require().Equal(err, surrealdb.ErrNoRow)
+				wg.Done()
+			}(i)
+		}
+	})
+	wg.Wait()
+
+	s.Run(fmt.Sprintf("%d Concurrent create", totalcoroutine), func() {
+		for i := 0; i < totalcoroutine; i++ {
+			wg.Add(1)
+			go func(j int) {
+				_, err := s.db.Create(fmt.Sprintf("users:%d", j), user)
+				s.Require().NoError(err)
+				wg.Done()
+			}(i)
+		}
+		wg.Wait()
+	})
+
+	s.Run(fmt.Sprintf("%d 10 Request Concurrent select exist", totalcoroutine), func() {
```

Comment:
Same comment as above on naming the test case scenario properly. I don't the 10 requests is a typo

Reply:
done